tests/main/cgroup-devices/task.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/sh -ex
+echo "Install a test service"
+snap pack test-snapd-service
+
+# Because this service is of type "simple" it is considered "ready" instantly.
+# In reality the process needs to go through snap "run" chain to be really
+# ready. As a workaround, touch a "remove-me" file that is removed by the
+# service on startup, restart the service and the wait for the file to
+# disappear.
+mkdir -p /var/snap/test-snapd-service/common
+touch /var/snap/test-snapd-service/common/remove-me
+snap install --dangerous ./test-snapd-service_1.0_all.snap
+# Wait for the service to really be alive and running. Otherwise the "main pid"
+# will be still tracking snap-run-confine-exec chain and be unreliable.
+for _ in $(seq 5); do
+	if [ ! -e /var/snap/test-snapd-service/common/remove-me ]; then
+		break
+	fi
+	sleep 1
+done
+
+echo "Extract the PID of the main process tracked by systemd"
+# It would be nicer to use "systemctl show --property=... --value" but it doesn't work on older systemd.
+pid=$(systemctl show snap.test-snapd-service.test-snapd-service.service --property=ExecMainPID | cut -d = -f 2)
+
+echo "Extract the device cgroup of the main process"
+initial_device_cgroup=$(grep devices < "/proc/$pid/cgroup" | cut -d : -f 3)
+
+# Initially, because there are no udev tags corresponding to this application,
+# snap-confine is not moving the process to a new cgroup. As such the service
+# runs in the cgroup created by systemd, which varies across version of systemd.
+case "$initial_device_cgroup" in
+    /)
```

Comment:
AMZN2?

Reply:
Indeed, check the commit message here for details https://github.com/snapcore/snapd/pull/7218/commits/5dd2b2cf39ebe03603c76629e1fe0494b79afaf5